src/recordSchema.ts

```diff
@@ -0,0 +1,63 @@
+
+import { Dictionary, expandUrl, loadField, idmapFieldsUnionOfNoneTypeOrArrayOfRecordFieldLoader, typeDSLEnumd9cba076fca539106791a4f46d198c7fcfbdb779Loader2, LoadingOptions, Saveable, ValidationException } from './util/internal'
+
+export class RecordSchema extends Saveable {
+  fields?: any
+  type: any
+  extensionFields?: Dictionary<any>
+  loadingOptions?: LoadingOptions
+
+  constructor (type: any, fields?: any, extensionFields?: Dictionary<any>, loadingOptions?: LoadingOptions) {
+    super()
+    this.extensionFields = extensionFields ?? {}
+    this.loadingOptions = loadingOptions ?? new LoadingOptions({})
+    this.fields = fields
+    this.type = type
+  }
+
+  static override async fromDoc (doc: any, baseuri: string, loadingOptions: LoadingOptions, docRoot?: string): Promise<Saveable> {
+    const _doc = Object.assign({}, doc)
+    const errors: ValidationException[] = []
+    if ('fields' in _doc) {
+      try {
+        var fields = await loadField(_doc.fields, idmapFieldsUnionOfNoneTypeOrArrayOfRecordFieldLoader, baseuri, loadingOptions)
+      } catch (e) {
+        if (e instanceof ValidationException) {
+          errors.push(new ValidationException('the `fields` field is not valid because: ', [e]))
+        }
+      }
+    } else {
+      fields = undefined
```

Comment:
I think this is an error? There is no `let fields` accessible here, I think...

Reply:
See one above 😄